src/java/org/apache/cassandra/service/accord/serializers/WaitingOnSerializer.java

```diff
@@ -73,22 +68,17 @@ public static ByteBuffer serialize(TxnId txnId, WaitingOn waitingOn) throws IOEx
         if (waitingOn.appliedOrInvalidated != null)
             appliedOrInvalidatedLength = (txnIdCount + 63) / 64;
 
-        Timestamp executeAtLeast = waitingOn.executeAtLeast();
-        ByteBuffer out = ByteBuffer.allocate(1 + (executeAtLeast == null ? 0 : CommandSerializers.timestamp.serializedSize())
-                                             + TypeSizes.sizeofUnsignedVInt(keyCount) + TypeSizes.sizeofUnsignedVInt(txnIdCount)
+        ByteBuffer out = ByteBuffer.allocate(TypeSizes.sizeofUnsignedVInt(waitingOnLength)
+                                             + (waitingOn.appliedOrInvalidated == null ? 0 : TypeSizes.sizeof(waitingOnLength - appliedOrInvalidatedLength))
                                              + TypeSizes.LONG_SIZE * (waitingOnLength + appliedOrInvalidatedLength));
 
-        if (executeAtLeast == null) out.put((byte)0);
-        else
-        {
-            out.put((byte) 1);
-            CommandSerializers.timestamp.serialize(executeAtLeast, out);
-        }
-        VIntCoding.writeUnsignedVInt32(keyCount, out);
-        VIntCoding.writeUnsignedVInt32(txnIdCount, out);
+        VIntCoding.writeUnsignedVInt32(waitingOnLength, out);
         serialize(waitingOnLength, waitingOn.waitingOn, out);
         if (appliedOrInvalidatedLength > 0)
```

Comment:
I think we need to use `waitingOn.appliedOrInvalidated != null` here, as there seem to be cases where we can have empty appliedOrInvalidated length, in which case we will compute size differently. An alternative is of course, when computing size, to use `> 0` as a check.